src/frontend/src/common/components/library/data_subview/components/CreateNSTreeModal/components/SampleFiltering/components/LocationFilter/index.tsx

```diff
@@ -0,0 +1,107 @@
+import { PopperProps } from "@mui/material";
+import { DefaultMenuSelectOption, DropdownPopper } from "czifui";
+import { escapeRegExp } from "lodash";
+import {
+  StyledDropdown,
+  StyledFilterGroup,
+  StyledFilterGroupName,
+} from "../../style";
+
+export type LocationFilterType = {
+  namedLocations: NamedGisaidLocation[];
+  selectedLocation: NamedGisaidLocation | null;
+  setSelectedLocation: (location: NamedGisaidLocation | null) => void;
+};
+
+interface LocationFilterStyleProps {
+  fullWidth?: boolean;
+  showTitle?: boolean;
+}
+
+interface LocationFilterProps
+  extends LocationFilterType,
+    LocationFilterStyleProps {}
+
+interface AutocompleteState {
+  getOptionLabel: (option: NamedGisaidLocation) => string;
+  inputValue: string;
+}
+
+export const LocationFilter = ({
```

Comment:
It looks like everything in here is a pretty close match to how the `LocationField` component in the metadata upload/edit tables works. I see there are a few differences and I wouldn't be surprised if it's hard to refactor the old `LocationField` component to generalize it, but I am worried about if we start to have a bunch of these similar-but-slightly-different versions of the location picker start to show up. Maybe just throw in some docs/comments here and in the other one pointing out that, if we keep building location pickers like this, we need to refactor and extract them into some shareable core component.

Reply:
Sounds good.  I think refactoring should wait until the 500+ sample project is complete.  I'll comment for now.